### ISSUE TITLE
File format differences

### ISSUE THREAD

**Issue by DavidMStraub** [DsixTools/python-smeftrunner#6]
@celis:

> 
> We also see some differences in the structure/formatting of the output card compared to the DsixTools.   We list them  here.  

 
- [ ] The Block SCALES is missing in the output card

- [x] blocks or entries with zero value not printed in the output card by python-smeftrunner.  DsixTools  prints the output card including zero entries/blocks
 
- [x] the ordering of the blocks differs in the output card compared to the one produced with DsixTools.  
  
- [x] The last block of the DsixTools output Block OPTIONS is missing.  This seems natural to us since this block is tightly linked to DsixTools functionality.

**Comment by DavidMStraub** [comment]
The issue with the ordering (SM parameters came after WCs) was fixed in the above commit I believe.

The block SCALES has to be added, right.

Concerning zero entries: when interpreting input files, you are treating missing entries as zero, right? So what's the rationale of printing out a large number of zero entries in the output? Isn't this a waste of disk space?

Concerning the OPTIONS block: this refers to the options of the Mathematica package and does not have any correspondence in the Python version - at least at the moment. At a later stage, one can add command-line scripts that mimick more closely the functionality of the Mathematica package and have several options that can then be added to the OUTPUT file.

**Comment by celis** [comment]
1-"The issue with the ordering (SM parameters came after WCs) was fixed in the above commit I believe.

The block SCALES has to be added, right."

Perfect! 

2-"Concerning zero entries: when interpreting input files, you are treating missing entries as zero, right?"

Right

3-"So what's the rationale of printing out a large number of zero entries in the output? Isn't this a waste of disk space?"

Yes you are right, your solution is more economical in disk space.    It should not cause any problem anyway because of (2.) above.  

4-"Concerning the OPTIONS block: this refers to the options of the Mathematica package and does not have any correspondence in the Python version - at least at the moment. At a later stage, one can add command-line scripts that mimick more closely the functionality of the Mathematica package and have several options that can then be added to the OUTPUT file."  

Ok!

**Comment by DavidMStraub** [comment]
In trying to implement the SCALES block I ran into a subtlety.

In the Mathematica version, there are two entries in the block: 1 is the UV scale, 2 is the EW scale.

However, in general, one is not forced to do a top-down evolution; one could also run bottom-up or whatever.

In addition, the UV scale plays a dual role in the original code: as the initial value for the RG evolution, and as a parameter the beta functions depend on explicitly (through terms of the form m_H^2/Lambda^2).

So in implementing this, I would interpret the entries of the block as

1 -> UV scale that enters the beta functions (but is not necessarily the starting point of the RG evolution)
2 -> the scale at which the following coefficients are given (which can be the EW scale, but could also be the high scale or something in between).

The (only?) drawback is that the initial scale is not specified, but this is not strictly necessary I guess.

Comments?

**Comment by DavidMStraub** [comment]
Comments? This is the last remaining issue.

**Comment by DavidMStraub** [comment]
I implemented this temporarily in 02df422d1976defe6a9393e15236cb125b970b80 as follows.

By default, the output will only contain

```
BLOCK SCALES
1 XXX # <- UV scale suppressing the operators
```

while the option `scale_out` can be manually specified which will result in

```
BLOCK SCALES
1 XXX # <- UV scale suppressing the operators
2 YYY # scale_out
```

**Comment by celis** [comment]
Hi,   

the points you make are valid.  

I guess the best thing is that you implement this aspect in the way you find it more convenient for the users. 

Regards!
